src/coreclr/interpreter/stackmap.cpp

```diff
@@ -28,10 +28,12 @@ InterpreterStackMap* GetInterpreterStackMap(ICorJitInfo* jitInfo, CORINFO_CLASS_
     InterpreterStackMap* result = nullptr;
     if (!t_sharedStackMapLookup)
         t_sharedStackMapLookup = dn_simdhash_ptr_ptr_new(0, nullptr);
+    assert(t_sharedStackMapLookup);
+
     if (!dn_simdhash_ptr_ptr_try_get_value(t_sharedStackMapLookup, classHandle, (void **)&result))
     {
         result = new InterpreterStackMap(jitInfo, classHandle);
-        dn_simdhash_ptr_ptr_try_add(t_sharedStackMapLookup, classHandle, result);
+        assert(dn_simdhash_ptr_ptr_try_add(t_sharedStackMapLookup, classHandle, result) == DN_SIMDHASH_ADD_INSERTED);
```

Comment:
This is not a safe use of assert. The assert macros we use in the runtime do not reliably run the contents of the assert on debug builds.

Reply:
Instead you should have an if statement, and if it fails call NOMEM() which will throw.